src/domain/community/community/community.resolver.queries.ts

```diff
@@ -0,0 +1,25 @@
+import { Inject } from '@nestjs/common';
+import { Profiling } from '@src/common/decorators';
+import { Args, Query, Resolver } from '@nestjs/graphql';
+import { UUID_NAMEID } from '@domain/common/scalars';
+import { CommunityService } from './community.service';
+import { ICommunity } from './community.interface';
+
+@Resolver()
+export class CommunityResolverQueries {
+  constructor(
+    @Inject(CommunityService) private communityService: CommunityService
+  ) {}
+
+  @Query(() => ICommunity, {
+    nullable: false,
+    description:
+      'A community. If no ID is specified then the first community is returned.',
+  })
+  @Profiling.api
+  async community(
+    @Args('ID', { type: () => UUID_NAMEID }) ID: string
```

Comment:
The community `name` is only unique when in the scope of ecoverse or  challenge or both. Either use only UUID on specify input type that accept ecoverseId and challengeId.